docs/RFCS/store_pool.md

```diff
@@ -0,0 +1,79 @@
+- Feature Name: Store Pool
```

Comment:
If it only was a repository for the health of all stores, then sure, but once we add any other logic, it doesn't make sense.